vllm_spyre/platform.py

```diff
@@ -218,16 +218,21 @@ def validate_request(
 
         if envs_spyre.VLLM_SPYRE_USE_CB:
             # For continuous batching, check if the request is within the max
-            # context length.
-            # The V1 engine will check the prompt length, but not the prompt +
-            # max_tokens.
-            if (prompt_len + max_tokens
+            # context length. This needs to take the padded prompt length
+            # into account.
+
+            # ceil division to pad to next block boundary
+            n = prompt_len
+            d = 64  # hardcoded AIU Spyre block size
+            prompt_padding_len = ((n + d - 1) // d) * d
+            if (prompt_padding_len + max_tokens
                     > envs_spyre.VLLM_SPYRE_MAX_CONTEXT_LENGTH):
```

Comment:
bit of a tangential question but how does the `validate_request` get access to vllm configs? Thinking out loud when we want to remove these global env variables

Reply:
possibly add a `cls` config variable that gets assigned the `vllm_config` at `check_and_update_config`? 🤔

Reply:
hmmmmmm

`vllm.get_config()`?

Reply:
uhhh vllm/config.py already imports vllm.platforms :( :(